src/main/java/com/contrastsecurity/sdk/scan/CodeArtifactImpl.java

```diff
@@ -0,0 +1,61 @@
+package com.contrastsecurity.sdk.scan;
+
+import java.time.Instant;
+import java.util.Objects;
+
+/** Implementation of the {@link CodeArtifact} resource. */
+final class CodeArtifactImpl implements CodeArtifact {
```

Comment:
I'm sure there are some auto-delegation libraries that could potentially remove some of this boiler plate, one that comes to mind is: https://github.com/ryandens/auto-delegate
I've heard good things about this dev too!

Reply:
I would love to use this generator, but it doesn't fit this use case. It generates code for delegating to an instance in the same type hierarchy e.g.

```
class MySet implements Set {
}
```